addons/mail/static/src/components/message/message.xml

```diff
@@ -177,7 +177,7 @@
                             </ul>
                         </t>
                     </div>
-                    <t t-if="message.subject and threadView and threadView.thread and (threadView.thread.mass_mailing or [env.messaging.inbox, env.messaging.history].includes(threadView.thread))">
+                    <t t-if="message.subject and message.originThread and message.originThread.name.trim() !== message.cleanedSubject and threadView and threadView.thread and (threadView.thread.mass_mailing or [env.messaging.inbox, env.messaging.history].includes(threadView.thread))">
```

Comment:
I don't like that we have to trim the thread name here and add name default value "". I think we will also need a "cleanedName" from thread name, that is cleaned up the same way as the subject, and that we can compare directly one with each other.

Reply:
@seb-odoo 
can we return boolean from here?
if the cleanedSubject and cleanedName are not satisfying the condition. so, that we don't need a comparison in the XML template.
rather than we will do a comparison on field compute?

Reply:
Yes, we can make a compute with a name such as `isSubjectEqualThreadName` if you prefer having condition in model instead of template.